llvm/test/Transforms/LoopVectorize/as_cast.ll

```diff
@@ -0,0 +1,21 @@
+; RUN: opt -passes=loop-vectorize %s
+; ModuleID = '<bc file>'
+target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128"
+target triple = "x86_64-unknown-linux-gnu"
```

Comment:
Does this test depend on the X86 backend? Can you remove this and instead use the force-vector-width flag? Also please check the output.

Reply:
It doesn't depend on the X86 backend, I've set the right flags to reproduce it, thank you. I've added some `CHECK`s, please let me know if you think this is enough or if I should add more `CHECK` lines